client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatchStatusDateParser.java

```diff
@@ -22,28 +22,28 @@
 import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.time.DateFormatter;
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.index.mapper.DateFieldMapper;
-import org.joda.time.DateTime;
-import org.joda.time.DateTimeZone;
 
 import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
 
 public final class WatchStatusDateParser {
 
-    private static final DateFormatter FORMATTER = DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER;
+    private static final DateFormatter FORMATTER = DateFormatter.forPattern("strict_date_optional_time||epoch_millis");
```

Comment:
I can see the same one used in DateFieldMapper - should we create a "default_date_time_formatter" in `DateFormatters` ?